apps/cowswap-frontend/src/modules/trade/hooks/useNavigateOnCurrencySelection.ts

```diff
@@ -53,18 +54,32 @@ export function useNavigateOnCurrencySelection(): CurrencySelectionCallback {
       const { inputCurrencyId, outputCurrencyId } = state
       const tokenSymbolOrAddress = resolveCurrencyAddressOrSymbol(currency)
 
-      const targetInputCurrencyId = field === Field.INPUT ? tokenSymbolOrAddress : inputCurrencyId
-      const targetOutputCurrencyId = field === Field.INPUT ? outputCurrencyId : tokenSymbolOrAddress
+      const isInputField = field === Field.INPUT
+      const targetInputCurrencyId = isInputField ? tokenSymbolOrAddress : inputCurrencyId
+      const targetOutputCurrencyId = isInputField ? outputCurrencyId : tokenSymbolOrAddress
       const areCurrenciesTheSame = targetInputCurrencyId === targetOutputCurrencyId
 
+      /**
+       * Change network to the token network only when select a sell token
+       * Because we allow to sell only tokens from supported networks
+       */
+      const targetChainId = isInputField ? currency?.chainId || chainId : chainId
```

Comment:
The most important change.
Change the network in URL (which triggers switching in the wallet) corresponding to the selected token.